src/dlmalloc.rs

```diff
@@ -24,42 +25,22 @@ pub struct Dlmalloc {
     trim_check: usize,
     least_addr: *mut u8,
     release_checks: usize,
+    phantom: PhantomData<S>,
 }
 
-unsafe impl Send for Dlmalloc {}
-
-pub const DLMALLOC_INIT: Dlmalloc = Dlmalloc {
-    smallmap: 0,
-    treemap: 0,
-    smallbins: [0 as *mut _; (NSMALLBINS + 1) * 2],
-    treebins: [0 as *mut _; NTREEBINS],
-    dvsize: 0,
-    topsize: 0,
-    dv: 0 as *mut _,
-    top: 0 as *mut _,
-    footprint: 0,
-    max_footprint: 0,
-    seg: Segment {
-        base: 0 as *mut _,
-        size: 0,
-        next: 0 as *mut _,
-        flags: 0,
-    },
-    trim_check: 0,
-    least_addr: 0 as *mut _,
-    release_checks: 0,
-};
+unsafe impl<S: System + Send> Send for Dlmalloc<S> {}
+
 
 // TODO: document this
-const NSMALLBINS: usize = 32;
-const NTREEBINS: usize = 32;
-const SMALLBIN_SHIFT: usize = 3;
-const TREEBIN_SHIFT: usize = 8;
+pub(crate) const NSMALLBINS: usize = 32;
+pub(crate) const NTREEBINS: usize = 32;
+pub(crate) const SMALLBIN_SHIFT: usize = 3;
+pub(crate) const TREEBIN_SHIFT: usize = 8;
 
 // TODO: runtime configurable? documentation?
-const DEFAULT_GRANULARITY: usize = 64 * 1024;
-const DEFAULT_TRIM_THRESHOLD: usize = 2 * 1024 * 1024;
-const MAX_RELEASE_CHECK_RATE: usize = 4095;
+pub(crate) const DEFAULT_GRANULARITY: usize = 64 * 1024;
+pub(crate) const DEFAULT_TRIM_THRESHOLD: usize = 2 * 1024 * 1024;
+pub(crate) const MAX_RELEASE_CHECK_RATE: usize = 4095;
```

Comment:
Why are all these changes (and `struct Segment` below) needed?

Reply:
Fixed